### ISSUE TITLE
6.22.0 removed all my PHPStorm windows on macOS 10.15.7

### ISSUE THREAD

**Issue by tangix** [lwouis/alt-tab-macos#948]
**Describe the bug**

After upgrading to 6.22.0 and even restart the Mac, all of my PHPStorm 2020.3.3 windows are missing out from AltTab.

**Steps to reproduce the bug**

1. Open AltTab
2. Open PHPStorm 2020.3 and a project
3. Use AltTab to switch between applications
4. PHPStorm projects not showing

Update:
Seems to be related to the use of Spaces. I normally run PHPStorm on Desktop 4 and my browser on Desktop 1 and then AltTab between them. Having both Chrome and PHPStorm on Desktop 1 and it works as before.
Downgraded to 6.21.2 and all is well again.

**Comment by lwouis** [comment]
@tangix I think this is the root cause: https://github.com/lwouis/alt-tab-macos/issues/885#issuecomment-839540866

**Comment by lwouis** [comment]
I will update the workaround to the following logic:

```
If the window is from a jetbrain app
  if it's a kAXStandardWindowSubrole
    show it
  else if it's a kAXDialogSubrole
    show it only if title is not empty
```

Hopefully this makes it workaround both:

* The empty title windows that we want to show (e.g. a project with no tab, opened from the welcome popup)
* The empty title non-windows that we don't want to show (e.g. a big auto-complete popup that's technically implemented by jetbrains as a macos window)

**Comment by tangix** [comment]
> @tangix I think this is the root cause: [#885 (comment)](https://github.com/lwouis/alt-tab-macos/issues/885#issuecomment-839540866)

Thanks for looking into this! Will check for a build and take it for a test-ride. 
Can only image the implications of trying to handle JetBrains' apps with multiple windows and also running a customized Java...

**Comment by lwouis** [comment]
> Thanks for looking into this! Will check for a build and take it for a test-ride.

[Here's a build](https://github.com/lwouis/alt-tab-macos/files/6464535/AltTab.app.zip) with the change described above. Could you let me know if it behaves great in all scenarios for you?

> Can only image the implications of trying to handle JetBrains' apps with multiple windows and also running a customized Java...

I wish they would fix this, instead of me working around their issues in AltTab. It's a big corp though, so me doing it bring a practical solution to people in the meanwhile (maybe forever since accessibility is probably not high on their radar, and the cost to change may be high, so they may never fix it)

**Comment by nmiticlevi9** [comment]
So far the problem I described yesterday is gone so great work, thank you, I'll keep this build, and will let you know if I encounter any more problems.

Thanks!!!

**Comment by tangix** [comment]
Quick testing shows no problems and works as well as 6.21.2 did!

**Comment by tangix** [comment]
> (maybe forever since accessibility is probably not high on their radar, and the cost to change may be high, so they may never fix it)
Off topic: Amen! Having a non-US keyboard layout (in my case Swedish) is hard on accessibility and also there are some serious bugs in handling our AltGr-key.

**Comment by gingerr** [comment]
Thank you for this fix and #885 ! 👍
